microceph/ceph/config.go

```diff
@@ -3,15 +3,144 @@ package ceph
 import (
 	"context"
 	"database/sql"
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/canonical/microceph/microceph/api/types"
 	"github.com/canonical/microceph/microceph/common"
 	"github.com/canonical/microceph/microceph/database"
 )
 
+type ConfigExtras struct{
+	Who     string // Ceph Config internal <who> against each key
+	Daemons []string // Daemons that need to be restarted post config change.
+	Regexp  string // Regular Expression to check value against key
+	Cb      *func(s common.StateInterface, c types.Config)
+}
+
+type ConfigItem map[string]ConfigExtras
+
+// Check if certain key is present in the map.
+func (c ConfigItem) isKeyPresent(key string) bool {
+	if _, ok := c[key]; !ok {
+        return false
+    }
+
+	return true
+}
+
+// Return keys of the given set
+func (c ConfigItem) Keys() (keys []string) {
+    for k := range c {
+        keys = append(keys, k)
+    }
+    return keys
+}
+
+func GetConfigTable() ConfigItem {
+	return ConfigItem{
+		"public_network":  {"global", []string{"mon", "osd"}, "", nil},
+		"cluster_network": {"global", []string{"osd"}, "", nil},
+	}
+}
+// Instantiation of config table.
+var configTable = GetConfigTable()
+
+// Struct to get Config Items from config dump json output.
+type ConfigDump []struct{
+	Section string
+	Name    string
+	Value   string
+}
+
+func RestartCephDaemon(daemon string) error {
+	return snapReload(daemon)
+}
+
+func SetConfigItem(s common.StateInterface, c types.Config) error {
+	args := []string{
+		"config",
+		"set",
+		configTable[c.Key].Who,
+		c.Key,
+		c.Value,
+	}
+
+	_, err := processExec.RunCommand("ceph", args...)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func GetConfigItem(s common.StateInterface, c types.Config) (types.Config, error) {
+	var err error
+	args := []string{
+		"config",
+		"get",
+		configTable[c.Key].Who,
+		c.Key,
+	}
```

Comment:
the other calls of `ceph config` in here seem to be using the `json-pretty` format

Reply:
the non-json output is just the value of the key that we inquire for, doing json/json-pretty gives a newline character and the value on the next line. I think we can skip the extra steps and just pump whatever value ceph gives us.